packages/wallet/ui/src/components/WalletConnection.jsx

```diff
@@ -67,8 +79,70 @@ const WalletConnection = ({
       },
     }).catch(rethrowIfNotCancelled);
 
+    /** @type {Map<string,[ReturnType<typeof makeCapTP>, number]>} */
+    const dappToConn = new Map();
+    const {
+      localStorage: storage,
+      addEventListener,
+      removeEventListener,
+    } = window; // WARNING: ambient
+
+    function handleStorageMessage(key, newValue) {
```

Comment:
a lot of logic in here that's not UI concern. consider extracting some to a utility function. that could be unit tested.

Reply:
concur

Reply:
Pulled this into a separate function. I don't know how much of this code is just placeholder/PoC until after we have the smart wallet bridge fully working, so I won't bother writing tests yet.